src/keygen/participant.rs

```diff
@@ -653,6 +721,7 @@ mod tests {
         }
         Ok(())
     }
+    // XXX should this test be re-written to use `Participant<KeygenParticipant>`?
```

Comment:
Yes you are right. I was hoping to avoid the duplicate code across the various participants, but yes, this change would break the abstraction boundaries (it might still be worth having such a test under `Participant` while keeping this one around as well)